documents/vci-front-end.md

```diff
@@ -0,0 +1,117 @@
+| `routes`  |                       | This object contains a collection of routes within the application, where each route is a sequence of steps leading the user through a process or workflow. Routes are how different pages are interconnected.                                                                                                                                               |
+|           | `id`                  | (optional) A unique identifier for a particular route; if omitted, the default route will be used. It is utilized mainly for internal routing logic.                                                                                                                                                                                                         |
+|           | `vpDefinitionId`      | (optional) A specific identifier that fetches the corresponding `presentation_definition` from the agent, dictating how presentations are processed and verified.                                                                                                                                                                                            |
+|           | `steps`               | This array defines a series of sequential steps for a route. Each step contains an `id` that matches a page configuration, an `operation` that dictates the action (such as "navigate" or "execute"), and optionally, `nextId` to indicate the subsequent step. The `isDefaultRoute` boolean flag sets the default pathway when the application initializes. |
+
+here are a few important note on these configurations:
+- We're supporting a wide range of options that are not necessarily used in all of our config samples. If you have a special case, be sure to read the interfaces behind the config.
+- It's necessary to populate the form in `infoRequest` page.
+- You can modify your flow to choose what suits your purpose. this can be achieved with modifying the `routes` property in your config json.
+- If your config contains new images, be sure to copy them in the public folder of this module
+
+#### Form configuration
+Here is a more in-depth look at the form configuration. A form is a list of lists. This is done to enable you to have multiple entries in a line.
+Each input entry can be defined with the following interface:
+```ts
+export interface DataFormElement {
+    id: string
+    key: string
+    type: HTMLInputTypeAttribute
+    required?: boolean
+    readonlyWhenAbsentInPayload?: boolean
+    defaultValue?: FormFieldValue
+    label?: string
+    labelUrl?: string
+    readonly?: boolean
+    customValidation?: string
+    display?: {
+        checkboxBorderColor?: string
```

Comment:
Hmzz, this really should have become a checkbox object, but that is out of the scope of this PR